cdap-ui/cypress/integration/pipeline.runtimeargs.spec.ts

```diff
@@ -48,21 +59,22 @@ const SOURCE_PATH_VAL = 'file:/tmp/cdap-ui-integration-fixtures/airports.csv';
     getArtifactsPoll(headers);
   });
 
-  it('should be successful for preview', () => {
+  it('and providing wrong runtime arguments should fail the pipeline preview.', () => {
     // Go to Pipelines studio
     cy.visit('/pipelines/ns/default/studio');
-    cy.url().should('include', '/studio');
+    cy.window().then((window) => {
+      skipPreviewTests = window.CDAP_CONFIG.hydrator.previewEnabled !== true;
+    });
+    if (skipPreviewTests) {
+      skip();
+    }
     cy.upload_pipeline(
       'pipeline_with_macros.json',
       '#pipeline-import-config-link > input[type="file"]'
-    ).then(subject => {
-      expect(subject.length).to.be.eq(1);
-    });
-  });
-
-  it('and providing wrong runtime arguments should fail the pipeline preview.', () => {
+    );
+    cy.wait(10000);
```

Comment:
We were trying to avoid explicit wait statements like these, right? You can use `cy.get('[title="Airport_source"').should('exist');` to check if the pipeline is deployed like we do in other tests?

Reply:
I agree we need to avoid the wait times but this is uploading a json in pipeline studio and we don't have a fail-proof way to find if its complete.

Reply:
Won't `cy.get('[title="Airport_source"').should('exist'); ` do this? We're testing if title in source node is present, which we only see after the pipeline is uploaded and ready in studio?